### ISSUE TITLE
Capture gui closing after writing and selecting another tool

### ISSUE THREAD

**Issue by delegadissim** [flameshot-org/flameshot#3359]
### Flameshot Version

Flameshot v11.0.0 (11.0.0-2 Debian)
Compiled with Qt 5.15.2

### Installation Type

Flatpak from Flathub

### Operating System type and version

Linux ti2-05 5.19.0-42-generic #43~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Apr 21 16:51:08 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### Description

Onece you've selected the area you want to capture and you write something (using the text feature), if you want to select another tool (clicking on it or with shortcuts) the capture closes. 

### Steps to reproduce

1. run flameshot gui
2. select the text tool
3. write something 
4. select another tool
5. the capture closes itself and you can't do anything

### System Information

Linux ti2-05 5.19.0-42-generic #43~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Apr 21 16:51:08 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

Ubuntu mate

**Comment by JoelValet** [comment]
Happening to me too, specs from neofetch

```            .-/+oossssoo+/-.               XPS-13-9320 
        `:+ssssssssssssssssss+:`           --------------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 22.04.3 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: XPS 9320 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 6.2.0-33-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 5 hours, 51 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2064 (dpkg), 20 (flatpak), 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: zsh 5.8.1 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080, 1920x1080 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: GNOME 42.9 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Mutter 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Adwaita 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Yaru-blue [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Papirus [GTK2/3] 
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: 12th Gen Intel i7-1260P (16) @  
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel Alder Lake-P 
      -+sssssssssssssssssyyyssss+-         Memory: 11753MiB / 31694MiB 
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.              ```

**Comment by mmahmoudian** [comment]
Duplicate of:

- #3265
- #3232
- #3120
- #2860

This issue is fixed in #2369
